osquery/tables/cloud/CMakeLists.txt

```diff
@@ -5,7 +5,9 @@
 # the LICENSE file found in the root directory of this source tree.
 
 function(osqueryTablesCloudMain)
-  generateOsqueryTablesCloud()
+  if(DEFINED PLATFORM_LINUX)
```

Comment:
We currently do not disable these kind of targets depending on the platform because they are EXCLUDE_FROM_ALL, so they are not built anyway.
Also Buck does the same thing and we mirrored it to be easy to sync back the changes to CMake.

That been said, I'm not saying this is necessarily the best thing to do, it has its pros and cons.
Pros:
- Keeps "compatibility" with Buck
- Always tests the highest amount of CMake code paths, to detect errors even when on other platforms.

Cons:
- CMake does more work for a platform that doesn't require that target
- IDE projects contains more targets than they require and could possibily slow down the IDE/loading of the project/s.
- IDE projects contains targets that won't build if individually asked to.

There's probably something else to be said, I've discussed this in the past with @alessandrogario, when we were developing the CMake build system; maybe lets open an issue about this?

Reply:
> We currently do not disable these kind of targets depending on the platform because they are EXCLUDE_FROM_ALL, so they are not built anyway.

This was not my experience. On windows it was trying to link the ec2 static lib, thus my motivation for disabling these.

Reply:
> > We currently do not disable these kind of targets depending on the platform because they are EXCLUDE_FROM_ALL, so they are not built anyway.
> 
> This was not my experience. On windows it was trying to link the ec2 static lib, thus my motivation for disabling these.

That's because something was linking osquery_tables_cloud on Windows, which you removed.
If it's EXCLUDE_FROM_ALL and nothing links to it/requires it, it won't be built.

Reply:
> Also Buck does the same thing and we mirrored it to be easy to sync back the changes to CMake.

This approach seems fine. But let's make an exception for this case here. I created #5664 to investigate re-enabling this in the future.